concepts/observability-strategy/opentelemetry-operator/README.md

```diff
@@ -0,0 +1,89 @@
+```
+
+### Instrumentation
+Annotate your app with either:
+```yaml
+annotations:
+  instrumentation.opentelemetry.io/inject-sdk: "kyma-system/kyma" #to inject env variables only
+  instrumentation.opentelemetry.io/inject-nodejs: "kyma-system/kyma" #to inject env variables and nodejs instrumentation library
+  instrumentation.opentelemetry.io/container-names: "myapp,myapp2" #to inject env variables and javaagent
+```
+
+### Findings
+
+The injection of environment variables does not only inject the destination for the exporters but also injects the typical OTEL resource attributes. Also it supports injecting trace propagation and sampling strategies. That is a very convenient feature to configure your workload declarative in case of it is already instrumented with the OTEL SDK.
+
+The auto-instrumentation was tested for java and nodejs. Here, we failed to get the nodejs instrumentation working at all, one application was coming up operational but without instrumentation active. Another one was even not starting up anymore.
+
+## Considerations, Risks, Potentials
+
+In the following several very relevant aspects are getting discussed in order to derive a architectural decision
+
+### Simplification of own abstraction logic
+
+The OpenTelemetry Operator is a well maintained project having an active community behind. The operator maps an otel-collector configuration in an actual deployment and manages the lifecycle of it. With that, the telemetry-operator does not need to manage concrete deployments/daemonsets and alike anymore but could rely on the intermediate abstraction and with that could get a simplified implementation. New features in the underlying operator can be leveraged immediately in the layer above.
+
+However, the current functionality of the OpenTelemetry Operator in regards to deployments is very limited. It mainly maps the collector definition 1:1 to the underlying k8s resource. Actual logic which will simplify more complex scenarios are not covered, also simple scenarios like adding a batch processor in combination with a memory-ballast extension with settings fitting to the actual resource settings are not covered. So there is simplification in pure resource handling mainly.
+
+### Guarantees for custom instances
+
+As the OTEL operator will be a feature for users of the cluster, also the lifecycle of users collectors will be managed in regards to image updates (otherwise there is not much meaning in having the feature). Here, the question on the supported image arises. Is it just a plain base image with basic receivers/exporters or the full contrib image? If it ships the full contrib image, there cannot be given any guarantees an updates, it will be best effort only relying on the upstream project fully, it might break your collector. Also providing a hardened image from security perspective will be very challenging as there will be many 3party-dependencies included having no knowledge about actual usage.
```

Comment:
I tried to split the long section into several sentences for clarity, but might have misunderstood things.
Not sure I understand the last sentence at all 😆   

```suggestion
Because the OpenTelemetry Operator will be a feature for users of the cluster, also the lifecycle of the users' collectors will be managed in regards to image updates (otherwise, the feature is pretty meaningless). Here, the question on the supported image arises: 
Is it just a plain base image with basic receivers/exporters, or the full contrib image?
- If it ships the full contrib image, there cannot be given any guarantees and updates; it will be best-effort only, relying on the upstream project fully. It might break your collector. Also, providing a hardened image from security perspective will be very challenging as there will be many third-party dependencies included having no knowledge about actual usage.
```

Is it "and updates" or "on updates"?